stdlib/public/runtime/SwiftObject.mm

```diff
@@ -609,7 +609,8 @@ static bool isBridgeObjectTaggedPointer(void *object) {
   // bit set.
   SWIFT_MUSTTAIL return objcRetainAndReturn(object);
 #else
-  return swift_retain(static_cast<HeapObject *>(objectRef));
+  swift_retain(static_cast<HeapObject *>(objectRef));
+  return object;
```

Comment:
A comment here to avoid this pitfall would be nice.  I feel like we would be tempted to do this again.

Reply:
Good point.